### PR TITLE
Add initial attestation measurement calculation

### PR DESCRIPTION
**Pull request by conradgrobler**
This is the first step towards calculating the expected measurement in the attestation report from the Stage 0 firmware image and the enclave binary.

See #3486 for more details.

### CODE REVIEW

File: snp_measurement/src/stage0.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+use anyhow::Context;
+use log::{debug, info};
+use sha2::{Digest, Sha256};
+use std::path::PathBuf;
+use x86_64::PhysAddr;
+
+/// The address of the first byte after the end of the firmware image.
+///
+/// The firmware image gets loaded just below the 4GiB boundary.
+pub const FIRMWARE_TOP: PhysAddr = PhysAddr::new(0x1_0000_0000);
```
Comment (andrisaar) [inline snp_measurement/src/stage0.rs:26]: The BIOS gets shadowed in the low memory as well. I don't know exactly how QEMU handles this, but some other VMMs just copy the first 128K of memory to 1M-128K.

File: snp_measurement/src/stage0.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+pub fn load_stage0(stage0_rom_path: PathBuf) -> anyhow::Result<Vec<u8>> {
+    let stage0_bytes =
+        std::fs::read(&stage0_rom_path).context("couldn't load stage0 firmware ROM image")?;
+    debug!("Stage0 size: {}", stage0_bytes.len());
+
+    let mut stage0_hasher = Sha256::new();
+    stage0_hasher.update(&stage0_bytes);
+    let stage0_cecksum = stage0_hasher.finalize();
+    info!("Stage0 SHA256 hash: {}", hex::encode(stage0_cecksum));
+    Ok(stage0_bytes)
```
Comment (tiziano88) [inline snp_measurement/src/stage0.rs:38]: Personally I recommend the following:

```suggestion
    let mut stage0_hasher = Sha256::new();
    stage0_hasher.update(&stage0_bytes);
    let stage0_sha256_digest = stage0_hasher.finalize();
    info!("Stage0 digest: sha256:{}", hex::encode(stage0_sha256_digest));
    Ok(stage0_bytes)
```

In particular, use the term "hash" for the function, and "digest" for the value, and also prefix the digest with the algo when printing.

Especially here that there are other hashing algos floating around :)

**Review by tiziano88 — APPROVED** [review]

**Review by rbehjati — APPROVED** [review]

File: snp_measurement/src/stage0.rs (hunk 2 above)
Comment (conradgrobler) [reply snp_measurement/src/stage0.rs:38]: Done.

File: snp_measurement/src/stage0.rs (hunk 1 above)
Comment (conradgrobler) [reply snp_measurement/src/stage0.rs:26]: Done.

**Review by conradgrobler — COMMENTED** [review]
Thanks

**Review by andrisaar — APPROVED** [review]

**Pull request merged**